dask/dataframe/tests/test_dataframe.py

```diff
@@ -402,6 +423,58 @@ def test_reductions():
         assert eq(dds.mean(), pds.mean())
         assert eq(dds.nunique(), pds.nunique())
 
+def test_reductions_non_numeric_dtypes():
+    # test non-numric blocks
+
+    def check_raises(d, p, func):
+        assert raises(TypeError, lambda: getattr(d, func)().compute())
+        assert raises(TypeError, lambda: getattr(p, func)())
+
+    pds = pd.Series(['a', 'b', 'c', 'd', 'e'])
+    dds = dd.from_pandas(pds, 2)
+    assert eq(dds.sum(), pds.sum())
+    assert eq(dds.min(), pds.min())
+    assert eq(dds.max(), pds.max())
+    assert eq(dds.count(), pds.count())
+    check_raises(dds, pds, 'std')
+    check_raises(dds, pds, 'var')
+    check_raises(dds, pds, 'mean')
+    assert eq(dds.nunique(), pds.nunique())
+
+    for pds in [pd.Series(pd.Categorical([1, 2, 3, 4, 5], ordered=True)),
+                pd.Series(pd.Categorical(list('abcde'), ordered=True)),
+                pd.Series(pd.date_range('2011-01-01', freq='D', periods=5))]:
+        dds = dd.from_pandas(pds, 2)
+
+        check_raises(dds, pds, 'sum')
+        assert eq(dds.min(), pds.min())
+        assert eq(dds.max(), pds.max())
+        assert eq(dds.count(), pds.count())
+        check_raises(dds, pds, 'std')
+        check_raises(dds, pds, 'var')
+        check_raises(dds, pds, 'mean')
+        assert eq(dds.nunique(), pds.nunique())
+
+
+    pds= pd.Series(pd.timedelta_range('1 days', freq='D', periods=5))
+    dds = dd.from_pandas(pds, 2)
+    assert eq(dds.sum(), pds.sum())
+    assert eq(dds.min(), pds.min())
+    assert eq(dds.max(), pds.max())
+    assert eq(dds.count(), pds.count())
+
+    # ToDo: pandas supports timedelta std, otherwise dask raises:
+    # incompatible type for a datetime/timedelta operation [__pow__]
+    # assert eq(dds.std(), pds.std())
```

Comment:
Another thing to beware of here is that our computation of variance in dask.dataframe is still numerically imprecise.  This is [fixed in `dask.array`](https://github.com/ContinuumIO/dask/pull/297) but not yet ported over to `dask.dataframe`.  (cc @jcrist)

How does pandas compute the variance of datetimes?  Can we cast them as integers, compute the variance, and then recast as timedeltas?